### DISCUSSION TITLE
3.3.0

### DISCUSSION THREAD

**Discussion by mokun** [mars-sim/mars-sim#448] [General]
### A. Editions
3.3.0 are available in 4 editions : 
 1. Jar Swing Edition              : `3.3.0_swing_java11.jar`
 2. Jar Console Edition         : `3.3.0_console_java11.jar`
 3. Debian Swing Edition       : `3.3.0_swing_java11.deb`
 4. Debian Console Edition  : `3.3.0_console_java11.deb`

In two console editions above, either a beryx console or a SSH terminal can be used as the means of interacting with settlers, robots, settlement, vehicle, etc. in mars-sim. See wiki with regard to setting up [SSH console](https://github.com/mars-sim/mars-sim/wiki/Remote-Console-Connection).

In two Swing editions above, in addition to the beryx console, a rudimentary graphic interface based on Java Swing will be available for players to interact with the simulated world.

Most players prefers the 'Jar Swing Edition' as it runs on Windows, macOS and Linux, provided that Java Runtime or JDK has been preinstalled on your machine.

Download your preferred edition to your machine today by clicking on the filename at the `Assets` section at the bottom of this page.

> Note 1 : currently openjfx/JavaFX package is NOT required to be installed to run mars-sim.

### B. Java VM Support

This build was made to be Java 11-compliant using Oracle official Java 11 or above (OpenJDK 11 or above). 

We apologize that running mars-sim with Java version below 11 is NO LONGER being supported.

### C. Installing and Starting a Sim in the Debian Editions

In case of the two Debian Editions binaries, you may install them in your linux distro using a *debian installer* such as [GDebi](https://itsfoss.com/gdebi-default-ubuntu-software-center/) in Ubuntu, or using command line as shown [here](https://askubuntu.com/questions/40779/how-do-i-install-a-deb-file-via-the-command-line) as an example in Ubuntu.

To start a simulation, type `./mars-sim` inside `/usr/bin` to launch mars-sim.

> Note 2 : the player may use an editor to modify the placeholder text inside `./mars-sim` to match the downloaded version of the mars-sim in order for it to launch the simulation. File an ticket in the *Issue Tracker* if you run into problems.

### D. Starting a Sim in the Jar Editions

To start a new simulation, find the corresponding jar file. 
Open a console and in the folder to which the jarfile was previously downloaded , type the following : 
`> java -jar 3.3.0_swing_java11.jar`

Make sure you DO NOT place multiple builds of binary in the *same* folder or else java will pick the first jar in alphabetical order to start.

> Note 3 : The `new` switch is automatically added if no switches are entered. It will assume that you are interested in starting a new simulation.

If the player clicks on the jarfile to start the simulation, he/she will NOT be able to see the logs and the track trace.

It is HIGHLY recommend using a terminal / CLI to manually start the simulation. This way, the player may see the logs and stack trace in the terminal / CLI. 

### E. Loading a Sim

To load the previously saved simulation, open a console. In the folder where the jarfile is resided, type the following 
`> java -jar 3.3.0_swing_java11.jar load`

The `load` switch will tell mars-sim to open the */default* directory to look for the sim file called `default.sim`.

### F. Build Management

Make sure you DO NOT place multiple builds of binary in the *same* folder or directory since this could potentially cause java to look into the *same* classpath and pick an old build to start.

Always DELETE older builds or archive each build on separate folder.

You may tell which build version you are running as shown on the title bar of the application window.

<hr /><em>This discussion was created from the release <a href='https://github.com/mars-sim/mars-sim/releases/tag/3.3.0'>3.3.0</a>.</em>

**Comment by tinerfe1986** [comment]
Good Afternoon,

First of all, congratulations for the simulator, is a great idea, I have been following that project for a while and now Im testing 3.3.0 on Linux.
I found a few flaws that I don't know how to deal with,
- First, I have no control over missions like before, I can't select any rover, I cannot select where to explore, or I cannot cancel a mission
- I feel that I can only watch the simulation, sometimes, settlers get stuck somewhere, and just suffocate slowly, and I cannot do nothing, that new versions is quite frustrating.
Any advise how to sort this lack of control over the simulator?

Thank you

**Reply by mokun** [reply]
Hi @tinerfe1986 ,

Yes. It's a good idea to give more control over each mission. 

There must be a reason that it gets stuck. 

Is the vehicle broken and doesn't have the parts ? have the resources run out ? anyone having a medical emergency ? 

It will be great to capture some screenprints.

@bevans2000 ,

Shall we could add in a few buttons in the Mission Tool such as: 
1. Return home
2. Call for medical emergency
3. Request parts delivery
4. Skip the current waypoint, etc.. 

Is there a way to create an in-game bug submission system for a player to send log and screen prints directly to us ?

**Reply by bevans2000** [reply]
Yes we can certainly add ore control to Missions. Makes sense.
@mokun Didn't 3.3.0 still have problems with the AirLock logic? The AirLock is normally the problem with Settlers suffocating as they get stuck trying to re-enter.

**Reply by mokun** [reply]
>  Didn't 3.3.0 still have problems with the AirLock logic? The AirLock is normally the problem with Settlers suffocating as they get stuck trying to re-enter.

I don't remember if it was in 3.2.0 or 3.3.0 in which we did have some instances in which the occupants of the vehicle get stuck in the airlock and can't come back in.

Good that I don't see any now (do you ?) as we are looking forward to releasing 3.4.0.

**Reply by tinerfe1986** [reply]
> > Didn't 3.3.0 still have problems with the AirLock logic? The AirLock is normally the problem with Settlers suffocating as they get stuck trying to re-enter.
> 
> I don't remember if it was in 3.2.0 or 3.3.0 in which we did have some instances in which the occupants of the vehicle get stuck in the airlock and can't come back in.
> 
> Good that I don't see any now (do you ?) as we are looking forward to releasing 3.4.0.

that is exactly the problem, I could send you a savegame when that happens again If you would like to check deeply.

**Reply by tinerfe1986** [reply]
> Hi @tinerfe1986 ,
> 
> Yes. It's a good idea to give more control over each mission.
> 
> There must be a reason that it gets stuck.
> 
> Is the vehicle broken and doesn't have the parts ? have the resources run out ? anyone having a medical emergency ?
> 
> It will be great to capture some screenprints.
> 
> @bevans2000 ,
> 
> Shall we could add in a few buttons in the Mission Tool such as:
> 
> 1. Return home
> 2. Call for medical emergency
> 3. Request parts delivery
> 4. Skip the current waypoint, etc..
> 
> Is there a way to create an in-game bug submission system for a player to send log and screen prints directly to us ?

But this control was until version 3.2.0 I tried version 3.1.1 and I can still controlling the routes of the mission.
After 3.2.0 I cannot cancel even a mission the button simply does not work, and when I will chose a vehicles, simply there are no option to do it.

**Reply by bevans2000** [reply]
Let's add a dialog back to control the mission. Let's focus on reintroducing some common actions first and then expand into mission specific actions after that. The actions should also be available via the Console probably.

**Comment by mokun** [comment]
@tinerfe1986 

We've since released [v3.4.0](https://github.com/mars-sim/mars-sim/releases/tag/3.4.0). Do feel free to check it out. Thanks !
